pkg/operator/clientutil/clientutil.go

```diff
@@ -0,0 +1,148 @@
+package clientutil
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+
+	apps_v1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	k8s_errors "k8s.io/apimachinery/pkg/api/errors"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/validation"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var invalidDNS1123Characters = regexp.MustCompile("[^-a-z0-9]+")
+
+// SanitizeVolumeName ensures that the given volume name is a valid DNS-1123 label
+// accepted by Kubernetes.
+//
+// Copied from github.com/prometheus-operator/prometheus-operator/pkg/k8sutil.
+func SanitizeVolumeName(name string) string {
```

Comment:
Since this is copied verbatim we can probably ignore the implementation for review here.